bleak/__init__.py

```diff
@@ -26,63 +47,518 @@
+                return ``True`` only for the wanted device.
+            timeout:
+                Optional timeout to wait for detection of specified peripheral
+                before giving up. Defaults to 10.0 seconds.
+            **kwargs:
+                Additional arguments to be passed to the :class:`BleakScanner`
+                constructor.
+
+        Returns:
+            The :class:`BLEDevice` sought or ``None`` if not detected before
+            the timeout.
+
+        """
+        found_device_queue: asyncio.Queue[BLEDevice] = asyncio.Queue()
+
+        def apply_filter(d: BLEDevice, ad: AdvertisementData):
+            if filter_func(d, ad):
+                found_device_queue.put_nowait(d)
+
+        async with cls(detection_callback=apply_filter, **kwargs):
+            try:
+                async with async_timeout.timeout(timeout):
+                    return await found_device_queue.get()
+            except asyncio.TimeoutError:
+                return None
+
+
+class BleakClient:
+    """The Client Interface for Bleak Backend implementations to implement.
+
+    Args:
+        device_or_address:
+            A :class:`BLEDevice` received from a :class:`BleakScanner` or a
+            Bluetooth address (device UUID on macOS).
+        disconnected_callback:
+            Callback that will be scheduled in the event loop when the client is
+            disconnected. The callable must take one argument, which will be
+            this client object.
+        **timeout:
+            Timeout in seconds passed to the implicit ``discover`` call when
+            ``device_or_address`` is not a :class:`BLEDevice`. Defaults to 10.0.
+        **winrt:
+            Dictionary of WinRT/Windows platform-specific options.
+        **kwargs:
+            Additional keyword arguments for backwards compatibility.
+
+    .. warning:: Although example code frequently initializes :class:`BleakClient`
+        with a Bluetooth address for simplicity, it is not recommended to do so
+        for more complex use cases. There are several known issues with providing
+        a Bluetooth address as the ``device_or_address`` argument.
+
+        1.  macOS does not provide access to the Bluetooth address for privacy/
+            security reasons. Instead it creates a UUID for each Bluetooth
+            device which is used in place of the address on this platform.
+        2.  Providing an address or UUID instead of a :class:`BLEDevice` causes
+            the :meth:`connect` method to implicitly call :meth:`BleakScanner.discover`.
+            This is known to cause problems when trying to connect to multiple
+            devices at the same time.
+    """
+
+    def __init__(
+        self,
+        device_or_address: Union[BLEDevice, str],
+        disconnected_callback: Optional[Callable[[BleakClient], None]] = None,
+        *,
+        timeout: float = 10.0,
+        winrt: WinRTClientArgs = {},
```

Comment:
importing `WinRTClientArgs` is causing CI failures. we will just have to use `dict` as the type here.